OVERVIEW.md

```diff
@@ -0,0 +1,192 @@
+
+The number of errors from set-level validation may or may not have a relation to a number of rows, depending on how the check is coded. For example, for a check on uniqueness, the number of errors may equal the number of duplicated rows, the number of elements that are duplicated, or something else entirely.
+
+Of the built-in rule types, `CUSTOM_SQL_STATEMENT` is intended for set-level validation. That means that any check implemented using this rule type will produce output that can be expected from set-level validation.
+
+
+#### Rule Type `NOT_NULL`
+
+Violated if any row in the given column contains null.
+
+```yaml
+rules:
+  NOT_NULL_SIMPLE:
+    rule_type: NOT_NULL
+
+rule_bindings:
+  NOT_NULL:
+    entity_id: TEST_DATA
+    column_id: UNIQUE_KEY
+    row_filter_id: NONE
+    rule_ids:
+      - NOT_NULL_SIMPLE
+```
+
+Complete example: [Rule](configs/rules/base-rules.yml) and [rule binding](configs/rule_bindings/team-1-bindings.yml)
+
+
+#### Rule Type `NOT_BLANK`
+
+Violated if any row in the given column contains null.
+ 
+```yaml
+rules:
+  NOT_NULL_SIMPLE:
+    rule_type: NOT_NULL
+
+rule_bindings:
+  NOT_NULL:
+    entity_id: TEST_DATA
+    column_id: UNIQUE_KEY
+    row_filter_id: NONE
+    rule_ids:
+      - NOT_NULL_SIMPLE
+```
+
+Complete example: [Rule](configs/rules/base-rules.yml) and [rule binding](configs/rule_bindings/team-2-bindings.yml)
+
+
+#### Rule Type `REGEX`
+
+For `STRING` columns. This rule type allows the specification of a Python regular expression to define allowed patterns.
+
+The example shows a check for a valid email address.
+
+```yaml
+rules:
+  VALID_EMAIL:
+    rule_type: REGEX
+    dimension: conformance
+    params:
+      pattern: |-
+        ^[^@]+[@]{1}[^@]+$
+
+rule_bindings:
+  VALID_EMAIL:
+    entity_id: TEST_DATA
+    column_id: EMAIL
+    row_filter_id: NONE
+    rule_ids:
+      - VALID_EMAIL
+```
+
+Complete example: [Rule](configs/rules/base-rules.yml) and [rule binding](configs/rule_bindings/team-2-bindings.yml)
+
+#### Rule Type `CUSTOM_SQL_EXPRESSION`
+
+This rule type allows the specification of an SQL expression returning a boolean. The validation fails when the expression returns `TRUE`.
+
+This rule, contrary to `CUSTOM_SQL_STATEMENT` supports row-level validation (see preceding section on the differences between row-level and
+set-leve validation). This implies it has access to the row-level context of the table defined as "entity", meaning that all columns of the
+same row can be accessed.
+
+The below example of this rule performs a SELECT on a reference table to compare to a list of known currencies.
+
+The SQL condition can be parametrized using `custom_sql_parameters` (see the `CUSTOM_SQL_LENGTH` [example rule](configs/base-rules.yml) and [how it's used](configs/rule_bindings/team-2-rule-bindings.yml)). The SQL condition can also use an implicit parameter `$column`, to identify the column that the condition should be applied to.
```

Comment:
worth calling out that $column is resolved to the column-id field in the rule_binding

Reply:
Good point, thanks. I have elaborated the example a bit (I will include also an entity definition so I can explain exactly how it resolves).